modules/monitoring-configuring-remote-write.adoc

```diff
@@ -0,0 +1,176 @@
++
+[source,yaml]
+----
+        basicAuth:
+          username:
+            <usernameSecret>
+          password:
+            <passwordSecret>
+----
+Substitute `<usernameSecret>` and `<passwordSecret>` accordingly.
++
+* The following example configures client TLS authentication:
++
+[source,yaml]
+----
+         tlsConfig:
+           cert:
+              <certSecret>
+           keySecret:
+              <keySecret>
+----
+Substitute `<certSecret>` and `<keySecret>` accordingly.
++
+.. Add write relabel configuration values after the authentication credentials:
++
+[source,yaml]
+----
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: cluster-monitoring-config
+  namespace: openshift-monitoring
+data:
+  config.yaml: |
+    prometheusK8s:
+      remoteWrite:
+      - url: "https://remote.write.endpoint"
+        <endpoint_authentication_credentials>
+        <write_relabel_configs>
+----
++
+For `<write_relabel_configs>` substitute a list of write relabel configurations for metrics that you want to send to the remote endpoint.
++
+See the link:https://prometheus.io/docs/prometheus/latest/configuration/configuration/#relabel_config[Prometheus relabel_config documentation] for information about write relabel configuration options.
++
+The following example `cluster-monitoring-config` `ConfigMap` settings show how to:
++
+* forward a single metric called `my_metric`
++
+* configure basic authentication using the `remoteWriteAuth` secret with the `user` and `password` keys, which would contain the endpoint authentication credentials
++
+[source,yaml]
+----
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: cluster-monitoring-config
+  namespace: openshift-monitoring
+data:
+  config.yaml: |
+    prometheusK8s:
+      remoteWrite:
+      - url: "https://remote.write.endpoint"
+        writeRelabelConfig:
+        - source_labels: [__name__]
+          regex: 'my_metric'
+          action: keep
+        basicAuth:
+          username:
+            name: remoteWriteAuth
+            key: user
+          password:
+            name: remoteWriteAuth
+            key: password
+
+----
+
+. If required, configure remote write for the Prometheus instance that monitors user-defined projects by changing the `name` and `namespace` `metadata` values as follows:
++
+[source,yaml]
+----
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: user-workload-monitoring-config
+  namespace: openshift-user-workload-monitoring
+data:
+  config.yaml: |
+    prometheus:
+      remoteWrite:
+      - url: "https://remote.write.endpoint"
+        <endpoint_authentication_credentials>
+        <write_relabel_configs>
+----
++
+[NOTE]
+====
+The Prometheus config map component is called `prometheusK8s` in the `cluster-monitoring-config` `ConfigMap` object and `prometheus` in the `user-workload-monitoring-config` `ConfigMap` object.
+====
++
+The other config map components in the `data/config.yaml/prometheus/remoteWrite` section remain unchanged.
```

Comment:
What other components? Do we need to say this?

Reply:
Good point. I think this sentence can be omitted.